File: leaf-similar-trees/solution-2.go
/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val int
 *     Left *TreeNode
 *     Right *TreeNode
 * }
 */
func leafSimilar(root1 *TreeNode, root2 *TreeNode) bool {
    if root1 == nil && root2 == nil {
        return true
    }
    return leafSequence(root1) == leafSequence(root2)
}

func leafSequence(node *TreeNode) string {
    if node == nil {
        return ""
    }
    if node.Left == nil && node.Right == nil {
        return fmt.Sprintf("-%d", node.Val)
    }
    leftSeq := leafSequence(node.Left)
    rightSeq := leafSequence(node.Right)
    return fmt.Sprintf("%s%s", leftSeq, rightSeq)
}

File: average-of-levels-in-binary-tree/solution-2.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @return {number[]}
 */
var averageOfLevels = function(root) {
    const q = [root]
    const result = []
    while(q.length) {
        const size = q.length
        let sum = 0
        for(let i = 0; i < size; i++) {
            const node = q.shift()
            node.left && q.push(node.left)
            node.right && q.push(node.right)
            sum += node.val
        }
        result.push(sum / size)
    }
    return result
};